src/scenes/Home/TerrawareHomeView/PlantingSiteStats.tsx

```diff
@@ -0,0 +1,258 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { Box, Grid, Typography, useTheme } from '@mui/material';
+import { Icon } from '@terraware/web-components';
+import { useDeviceInfo } from '@terraware/web-components/utils';
+
+import Link from 'src/components/common/Link';
+import PlantingSiteSelector from 'src/components/common/PlantingSiteSelector';
+import { APP_PATHS } from 'src/constants';
+import { selectLatestObservation } from 'src/redux/features/observations/observationsSelectors';
+import { selectPlantingSites } from 'src/redux/features/tracking/trackingSelectors';
+import { useAppSelector } from 'src/redux/store';
+import strings from 'src/strings';
+import { PlantingSite } from 'src/types/Tracking';
+import useMapboxToken from 'src/utils/useMapboxToken';
+import { useDefaultTimeZone } from 'src/utils/useTimeZoneUtils';
+
+import StatsCardItem from './StatsCardItem';
+
+export const PlantingSiteStats = () => {
+  const { isDesktop } = useDeviceInfo();
+  const theme = useTheme();
+  const navigate = useNavigate();
+  const plantingSites = useAppSelector(selectPlantingSites);
+  const { token } = useMapboxToken();
+  const defaultTimeZone = useDefaultTimeZone();
+
+  const [selectedPlantingSiteId, setSelectedPlantingSiteId] = useState<number>();
+  const [selectedPlantingSite, setSelectedPlantingSite] = useState<PlantingSite>();
+
+  const observation = useAppSelector((state) =>
+    selectLatestObservation(state, selectedPlantingSiteId || -1, defaultTimeZone.get().id)
+  );
+
+  const geojson = {
+    type: 'FeatureCollection',
+    features: [
+      {
+        type: 'Feature',
+        geometry: selectedPlantingSite?.boundary,
+        properties: {
+          fill: theme.palette.TwClrBaseGreen300,
+          'fill-opacity': 0.2,
+          stroke: theme.palette.TwClrBaseGreen300,
+        },
+      },
+    ],
+  };
+  const geojsonString = encodeURIComponent(JSON.stringify(geojson));
+  const staticMapURL = `https://api.mapbox.com/styles/v1/mapbox/satellite-streets-v12/static/geojson(${geojsonString})/auto/580x360@2x?padding=100&access_token=${token}`;
+
+  const primaryGridSize = () => {
```

Comment:
I think this can be a memoized value instead of a function that is created with every render and called many times.

Reply:
I copied this pattern from another view, but you're right — will update